lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -2509,15 +2524,14 @@ void TFDeabstraction::evaluateAttributesAndDoPacking(
                                             dtypes))
         return diagnoseInvalidAttr("requires a TensorFlow value or an "
                                    "aggregate of TensorFlow values");
-      // Drop '$typeList' from the attribute name.
+      // Change "$typeList" to "$dtype".
       currentAttr.name = context.getIdentifier(
-          std::get<0>(argumentNameAndLowering));
-      // Replace the single metatype with a list of metatypes each
-      // representing a dtype.
-      auto *dtypeProto =
-          context.getProtocol(KnownProtocolKind::AccelerableByTensorFlow);
+          (std::get<0>(argumentNameAndLowering) + "$dtype").str());
+      // Replace the single metatype with a list of UInt32s each representing a
+      // TF_DataType.
       currentAttr.value = SymbolicValue::getArray(
-          dtypes, dtypeProto->getInterfaceType()->getCanonicalType(),
+          dtypes,
+          context.getUInt32Decl()->getDeclaredType()->getCanonicalType(),
```

Comment:
elsewhere in the code we seem to be using signed int32 (e.g. `sextOrTrunc(32)`) -- which one is correct? unsigned might be slightly better.

Consider documenting that (perhaps at `TFDataTypeAttribute`)

Reply:
Unsigned is correct. (TF_DataType is seen as `struct TF_DataType { let rawValue: UInt32 }` by swift. I will change everything to treat it as unsigned.